applications/FluidDynamicsApplication/custom_strategies/strategies/compressible_navier_stokes_explicit_solving_strategy_runge_kutta_4.h

```diff
@@ -0,0 +1,834 @@
+//
+//  Main authors:    Ruben Zorrilla
+//
+//
+
+#if !defined(KRATOS_COMPRESSIBLE_NAVIER_STOKES_EXPLICIT_SOLVING_STRATEGY_RUNGE_KUTTA_4)
+#define KRATOS_COMPRESSIBLE_NAVIER_STOKES_EXPLICIT_SOLVING_STRATEGY_RUNGE_KUTTA_4
+
+// System includes
+#include <functional>
+
+// External includes
+
+// Project includes
+#include "includes/define.h"
+#include "includes/global_variables.h"
+#include "includes/model_part.h"
+#include "solving_strategies/strategies/explicit_solving_strategy_runge_kutta_4.h"
+#include "utilities/math_utils.h"
+#include "utilities/parallel_utilities.h"
+
+// Application includes
+#include "fluid_dynamics_application_variables.h"
+
+namespace Kratos
+{
+
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+///@}
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+/** @brief Explicit solving strategy base class
+ * @details This is the base class from which we will derive all the explicit strategies (FE, RK4, ...)
+ */
+template <class TSparseSpace, class TDenseSpace>
+class CompressibleNavierStokesExplicitSolvingStrategyRungeKutta4 : public ExplicitSolvingStrategyRungeKutta4<TSparseSpace, TDenseSpace>
+{
+public:
+    ///@name Type Definitions
+    ///@{
+
+    /// The base class definition
+    typedef ExplicitSolvingStrategyRungeKutta4<TSparseSpace, TDenseSpace> BaseType;
+
+    /// The explicit builder and solver definition
+    typedef typename BaseType::ExplicitBuilderType ExplicitBuilderType;
+
+    /// The local vector definition
+    typedef typename TDenseSpace::VectorType LocalSystemVectorType;
+
+    /// Pointer definition of CompressibleNavierStokesExplicitSolvingStrategyRungeKutta4
+    KRATOS_CLASS_POINTER_DEFINITION(CompressibleNavierStokesExplicitSolvingStrategyRungeKutta4);
+
+    /// Local Flags
+    KRATOS_DEFINE_LOCAL_FLAG(SHOCK_CAPTURING);
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /**
+     * @brief Default constructor. (with parameters)
+     * @param rModelPart The model part of the problem
+     * @param ThisParameters The configuration parameters
+     */
+    explicit CompressibleNavierStokesExplicitSolvingStrategyRungeKutta4(
+        ModelPart &rModelPart,
+        Parameters ThisParameters)
+        : BaseType(rModelPart)
+    {
+        // Validate and assign defaults
+        ThisParameters = this->ValidateAndAssignParameters(ThisParameters, this->GetDefaultParameters());
+        this->AssignSettings(ThisParameters);
+    }
+
+    /**
+     * @brief Default constructor.
+     * @param rModelPart The model part to be computed
+     * @param pExplicitBuilder The pointer to the explicit builder and solver
+     * @param MoveMeshFlag The flag to set if the mesh is moved or not
+     */
+    explicit CompressibleNavierStokesExplicitSolvingStrategyRungeKutta4(
+        ModelPart &rModelPart,
+        typename ExplicitBuilderType::Pointer pExplicitBuilder,
+        bool MoveMeshFlag = false,
```

Comment:
actually you may even not need this parameter

Reply:
Eventually we might want to move the boundaries, this is why I kept it. OFC this would require to upgrade the element from the Eulerian to the ALE.